internal/cmd/plugin/destroy/destroy.go

```diff
@@ -0,0 +1,95 @@
+/*
+Copyright The CloudNativePG Contributors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package destroy implements a command to destroy an instances of a cluster and its associated PVC
+package destroy
+
+import (
+	"context"
+	"fmt"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/cloudnative-pg/cloudnative-pg/internal/cmd/plugin"
+)
+
+// Destroy implements the destroy subcommand
+func Destroy(ctx context.Context, clusterName, instanceID string, deletePVC bool) error {
+	// Check if the Pod exist
+	var pod v1.Pod
+	instanceName := fmt.Sprintf("%s-%s", clusterName, instanceID)
+
+	err := plugin.Client.Get(ctx, client.ObjectKey{Namespace: plugin.Namespace,
+		Name: instanceName}, &pod)
+
+	var volumes []v1.Volume
+	if err != nil {
+		return fmt.Errorf("instance %s not found in namespace %s", instanceName, plugin.Namespace)
+	}
+	volumes = pod.Spec.Volumes
+
+	// for the pod to be deleted it must either be owned by the cluster
+	if isOwnedByCluster(clusterName, pod.OwnerReferences) {
+		// Delete the Pod
+		err = plugin.Client.Delete(ctx, &pod)
+		if err != nil {
+			return fmt.Errorf("error deleting instance %s: %v", instanceName, err)
+		}
+	} else {
+		return fmt.Errorf("instance %s is not owned by cluster %s", instanceName, clusterName)
+	}
+
+	if deletePVC {
```

Comment:
I think that if we don't delete the PVC we should proceed to delete the owner reference from the PVCs otherwise they will get reconciled by the operator